arrows/core/tests/test_derive_metadata.cxx

```diff
@@ -0,0 +1,127 @@
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief test derivation of new metadata traits
+
+#include <arrows/core/derive_metadata.h>
+
+#include <vital/types/geo_point.h>
+#include <vital/types/geodesy.h>
+#include <vital/types/metadata.h>
+#include <vital/types/metadata_traits.h>
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <memory>
+#include <sstream>
+
+namespace kv = kwiver::vital;
+
+static kwiver::vital::metadata_traits meta_traits;
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+namespace {
+
+kwiver::vital::metadata_vector
+make_metadata()
+{
+  kv::metadata_sptr m1 = std::make_shared< kv::metadata >();
+
+  // Add the double-valued traits
+  std::map< kwiver::vital::vital_metadata_tag, double >
+  double_metadata{ { kv::VITAL_META_PLATFORM_HEADING_ANGLE, 60 },
+    { kv::VITAL_META_PLATFORM_PITCH_ANGLE, 3 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 1.2 },
+    { kv::VITAL_META_SENSOR_REL_AZ_ANGLE, 5 },
+    { kv::VITAL_META_SENSOR_REL_EL_ANGLE, 4 },
+    { kv::VITAL_META_SENSOR_REL_ROLL_ANGLE, 5 },
+    { kv::VITAL_META_PLATFORM_ROLL_ANGLE, 0.5 },
+    { kv::VITAL_META_SENSOR_VERTICAL_FOV, 30 },
+    { kv::VITAL_META_SENSOR_HORIZONTAL_FOV, 40 },
+    { kv::VITAL_META_SLANT_RANGE, 300 },
+    { kv::VITAL_META_TARGET_WIDTH, 100 } };
+
+  for( auto const& elem : double_metadata )
+  {
+    auto const& trait = meta_traits.find( elem.first );
+    m1->add( trait.create_metadata_item( elem.second ) );
+  }
+
+  // Add the geo point traits
+  auto const& sensor_loc_trait =
+    meta_traits.find( kv::VITAL_META_SENSOR_LOCATION );
+  m1->add( sensor_loc_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 12 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  auto const& frame_center_trait =
+    meta_traits.find( kv::VITAL_META_FRAME_CENTER );
+  m1->add( frame_center_trait.create_metadata_item(
+             kv::geo_point( kv::geo_point::geo_3d_point_t{ 0, 0, 0 },
+                            kv::SRID::lat_lon_WGS84 ) ) );
+
+  // Replicate these values except without the slant range field
+  auto const m2 = std::make_shared< kv::metadata >( m1->deep_copy() );
+  m2->erase( kv::VITAL_META_SLANT_RANGE );
+  return { m1, m2 };
+}
+
+} // end namespace
+
+class derive_metadata : public ::testing::Test
+{
+  void
+  SetUp()
+  {
+    auto meta = make_metadata();
+    size_t const frame_height{ 720 };
+    size_t const frame_width{ 1080 };
+
+    derived_metadata = kwiver::arrows::core::compute_derived_metadata(
+      meta, frame_width, frame_height );
+  }
+
+public:
+  kv::metadata_vector derived_metadata;
+};
+
+// ----------------------------------------------------------------------------
+TEST_F(derive_metadata, compute_derived)
```

Comment:
```suggestion
TEST_F( derive_metadata, compute_derived )
```

Reply:
Everywhere else I've seen uses no spaces for `TEST*`google test macros. I was fairly sure this was by convention.

Reply:
It *is* de-facto convention, but it's an exception with no real justification, and I'm not aware that anyone is really arguing for that exception; rather, it is an accident. It's also getting "corrected" as we start working to apply Uncrustify across the code base.

(On that note, as currently configured, Uncrustify wants to also add a space in `TEST (`, however that's considered a bug / misconfiguration.)

Reply:
I take that back, it looks like a mixed bag. But still, it appears `[ws.space]` suggests this style.

Reply:
Hmm, so it does. I'm not sure *why* that's the case, though, other than "inertia". Perhaps, like blank lines between namespaces, we should revisit that...

Reply:
If you view the raw `code-style.rst` files, there's a subitem under that stating this is mostly historical with no good justification, but that's not rendering properly.

Reply:
Ah... and it suggests nuking the exception if/when we start using automated formatting. Guess what we're starting to do 🙂.

(It *is* rendering correctly. It's a reST comment; the correct rendering is to not render, which it is correctly doing.)